tests/unit/test_scenario.py

```diff
@@ -0,0 +1,34 @@
+from unittest.mock import ANY
+
+import ops
+import pytest
+from scenario import Context, Container, Relation, State
+
+from charm import HexanatorCharm
+
+META = {
+    "name": "hexanator",
+    "requires": {
+        "ingress": {
+            "interface": "ingress"
+        }
+    },
+    "containers": {
+        "gubernator": {}
+    },
+
+}
+
+
+def test_startup():
+    ctx = Context(HexanatorCharm, meta=META)
+    container = Container(name="gubernator", can_connect=True)
+    relation=Relation(endpoint="ingress", interface="ingress", remote_app_name="ingress")
+    state = State( leader=True, relations=[relation], containers=[container])
```

Comment:
This may not have been merged when you wrote this, but we've moved to (frozen) sets for the state components. The code will still work - if an iterable is provided that's not a frozenset we'll convert it to a set (assuming each element can be hashed) - but the HEAD of the 7.0 branch should complain about the type being wrong.

```suggestion
    state = State(leader=True, relations={relation}, containers={container})
```

Reply:
I'll take the suggestion, however, that still doesn't keep pyright happy:

`"set[Container]" is incompatible with "FrozenSet[Container]"`

Reply:
Ah, then this is something 7.0 needs to fix. Sets are frozen when they are passed (just like everything else in the dataclass) but we don't want to force people to write `frozenset({....})`. If there was an explicit `__init__` I would have the type there be `Iterable` (or iterable-of-hashable) and the type of the attribute be frozenset. Since we don't have one, maybe it needs to be `set|frozenset`, which would make passing in happy, and probably the union would still provide errors if you try to change the frozen attribute. I'll try to look into this tomorrow or Monday.